gmso/formats/lammpsdata.py

```diff
@@ -654,13 +719,437 @@ def _get_ff_information(filename, unit_style, topology):
+    )
+    out_file.write("#\t" + "\t".join(param_labels) + "\n")
+    indexList = list(top.angle_types(filter_by=pfilter))
+    indexList.sort(
+        key=lambda x: (
+            x.member_types[1],
+            min(x.member_types[::2]),
+            max(x.member_types[::2]),
+        )
+    )
+    for idx, angle_type in enumerate(indexList):
+        out_file.write(
+            "{}\t{:7.5f}\t{:7.5f}\t#{}\t{}\t{}\n".format(
+                idx + 1,
+                _parameter_converted_to_float(
+                    angle_type.parameters["k"], base_unyts, cfactorsDict
+                ),
+                angle_type.parameters["theta_eq"]
+                .to(u.degree)
+                .value,  # write equilibrium values in degrees
+                *angle_type.member_types,
+            )
+        )
+
+
+from gmso.core.views import get_sorted_names
+
+
+def _write_dihedraltypes(out_file, top, base_unyts, cfactorsDict):
+    """Write out dihedrals to LAMMPS file."""
+    test_dihtype = top.dihedrals[0].dihedral_type
+    out_file.write(f"\nDihedral Coeffs #{test_dihtype.name}\n")
+    param_labels = map(
+        lambda x: f"{x} ({test_dihtype.parameters[x].units})",
+        test_dihtype.parameters,
+    )
+    out_file.write("#\t" + "\t".join(param_labels) + "\n")
+    indexList = list(top.dihedral_types(filter_by=pfilter))
+    index_membersList = [
+        (dihedral_type, get_sorted_names(dihedral_type))
+        for dihedral_type in indexList
+    ]
+    index_membersList.sort(key=lambda x: ([x[1][i] for i in [1, 2, 0, 3]]))
+    for idx, (dihedral_type, members) in enumerate(index_membersList):
+        out_file.write(
+            "{}\t{:8.5f}\t{:8.5f}\t{:8.5f}\t{:8.5f}\t# {}\t{}\t{}\t{}\n".format(
+                idx + 1,
+                *[
+                    _parameter_converted_to_float(
+                        dihedral_type.parameters[parameterStr],
+                        base_unyts,
+                        cfactorsDict,
+                    )
+                    for parameterStr in ["k1", "k2", "k3", "k4"]
+                ],
+                *members,
+            )
+        )
+
+
+def _write_impropertypes(out_file, top, base_unyts, cfactorsDict):
+    """Write out impropers to LAMMPS file."""
+    # TODO: Use any accepted lammps parameters
+    test_imptype = top.impropers[0].improper_type
+    out_file.write(f"\nImproper Coeffs #{test_imptype.name}\n")
+    param_labels = map(
+        lambda x: f"{x} ({test_imptype.parameters[x].units})",
+        test_imptype.parameters,
+    )
+    out_file.write("#\t" + "\t".join(param_labels) + "\n")
+    for idx, improper_type in enumerate(top.improper_types(filter_by=pfilter)):
+        out_file.write(
+            "{}\t{:7.5f}\t{:7.5f}\n".format(
+                idx + 1,
+                _parameter_converted_to_float(
+                    improper_type.parameters["k"], base_unyts, cfactorsDict
+                ),
+                improper_type.parameters["chieq"]
```

Comment:
```suggestion
                improper_type.parameters["phi_eq"]
```
Currently, in the `PeriodicImproperPotential` json, the three variables are `k` (energy), `n` (dimensionless) and `phi_eq` (angle). I think the `chieq` could be a typo. Also, why don't the `n` get written out somewhere?

Reply:
I guess, this is supposed to write out harmonic improper, in which case, we need to adjust the list in `_accepted_potential`

Reply:
Looking at the docs (https://docs.lammps.org/improper_style.html), looks like our `PeriodicImproper` is called `fourier` in lammps?

Reply:
Yeah so the naming is slightly weird, that's on of the reasons I was tagging things with LAMMPS as the potential template name.
As far as I can tell:
GMSO PeriodicImproperPotential = lammps modified improper_style CVFF (not sure if there's a 1-1 conversion here
GMSO HarmonicImproperPotential = 1/2* LAMMPS improper_style harmonic
GMSO FourierTorsionPotential != LAMMPS improper_style fourier (we would probably just need a separate template for it).

Reply:
The equations for PeriodicImproper and and lammps improper_style fourier, are similar, but missing the phi_eq term.

Reply:
So I am looking back here: https://github.com/mosdef-hub/mbuild/blob/main/mbuild/formats/lammpsdata.py. 
It seems like that the current mbuild version support LAMMPS improper `harmonic` (coming from`structure.impropers`) and `cvff` (coming from `structure.dihedrals` where `dihedral.improper == True`). So I think, in GMSO, that would map GMSO `harmonic` -->  LAMMPS `harmonic` and GMSO `periodic` --> LAMMPS `cvff`.

Reply:
Need to add a test for this, for some reason it's not being caught at all.